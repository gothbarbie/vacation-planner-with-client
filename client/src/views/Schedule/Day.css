.day {
  background: #F2F2F2;
  font-family: 'Avenir Next', sans-serif;
  font-weight: 600;
  font-size: 18px;
  color: #828282;
  border: 1px solid #F2F2F2;
}

.day:hover {
  border: 1px solid #80D1FF;
}

.day--weekend {
  color: #EB5757;
}

.day--empty {
  background: #fff;
}

.day--occupied {
  border: 1px solid #EB5757;
}

.day__date {
  margin: .75rem;
}

.day__people {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 3.5rem;
}

.day__status {
  font-family: 'Avenir Next', sans-serif;
  font-size: 13px;
  font-weight: 600;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 26px;
}

.day__status--occupied {
  color: #FFFFFF;
  background: #EB5757;
}
