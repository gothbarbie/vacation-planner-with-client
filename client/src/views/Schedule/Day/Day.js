// @flow

import React from 'react'
import styled from 'styled-components'

type props = {
  date?: number,
  occupied: boolean,
  people: [
    {
      name: string,
    },
  ],
  status?: 'string',
  today: boolean,
  weekend: boolean,
  empty: boolean,
}

const DayWrapper = styled.article`
  background-color: ${({ empty, theme }) =>
    empty ? theme.colors.white : theme.colors.grayLighter};
  font-family: ${({ theme }) => theme.typography.fontDefault};
  font-weight: 600;
  font-size: 1rem;
  border: 1px solid
    ${({ occupied, theme }) =>
      occupied ? theme.colors.dangerLight : theme.colors.grayLighter};

  color: ${({ theme, weekend }) =>
    weekend ? theme.colors.dangerLight : theme.colors.grayLight};

  border: ${({ theme, today }) =>
    today && '1px dashed ' + theme.colors.primary};

  &:hover {
    border: 1px solid ${({ theme }) => theme.colors.primary};
  }
`

const Date = styled.div`
  margin: 0.75rem;
`

const People = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 3.5rem;
`

const Status = styled.div`
  font-family: 'Avenir Next', sans-serif;
  font-size: 13px;
  font-weight: 600;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 26px;

  color: ${({ occupied, theme }) => occupied ? theme.colors.white : 'transparent'};
  background: ${({ occupied, theme }) => occupied ? theme.colors.dangerLight : 'transparent'};
`

const Day = ({
  date,
  empty,
  occupied,
  people,
  status,
  today,
  weekend,
}: props) => (
  <DayWrapper empty={empty} occupied={occupied} today={today} weekend={weekend}>
    <Date>{date}</Date>
    <People>
      {people &&
        people.map(person => {
          return <div>{person.name}</div>
        })}
    </People>
    <Status occupied={occupied}>{status}</Status>
  </DayWrapper>
)

export default Day
