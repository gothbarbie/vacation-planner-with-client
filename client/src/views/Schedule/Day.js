// @flow

import React from 'react'
import classnames from 'classnames'

import './Day.css'

type props = {
  date?: number,
  occupied: boolean,
  people: [{
    name: string
  }],
  status?: 'string',
  weekend: boolean,
  empty: boolean,
}

export default ({ date, empty, occupied, people, status, weekend }: props) => (
  <div
    className={classnames('day', {
      'day--weekend': weekend,
      'day--occupied': occupied,
      'day--empty': empty,
    })}>
    <div className="day__date">{date}</div>
    <div className="day__people">
      {people &&
        people.map(person => {
          return <div>{person.name}</div>
        })}
    </div>
    <div className={classnames('day__status', {
      'day__status--occupied': occupied,
    })}>{status}</div>
  </div>
)
