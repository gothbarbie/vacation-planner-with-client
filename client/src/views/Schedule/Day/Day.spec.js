import React from 'react'
import { shallow } from 'enzyme'
import Day from './Day'

describe('Day', () => {
  const day = shallow(<Day />)

  it('renders', () => {
    expect(day).toMatchSnapshot()
  })
})
