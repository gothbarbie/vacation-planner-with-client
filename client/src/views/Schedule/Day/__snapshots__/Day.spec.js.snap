// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renders 1`] = `
<div
  className="day"
>
  <div
    className="day__date"
  />
  <div
    className="day__people"
  />
  <div
    className="day__status"
  />
</div>
`;
